packages/client/src/components/Question/QuestionSection.tsx

```diff
@@ -80,6 +81,16 @@ const QuestionSection: FunctionComponent<SectionComponent> = (props) => {
     return null;
   }
 
+  // Prevent a bug when refreshing a checker without getDataNeed (only happens when first question remains unanswered)
+  if (
+    !getDataNeed(checker) &&
+    isActive &&
+    !isCompleted &&
+    checker.stack.length === 0
```

Comment:
can't we change this to a explaining const. 
```
const noDataNeedAndEmptyStack = !getDataNeed(checker) && isActive && !isCompleted && checker.stack.length === 0

if (noDataNeedAndEmptyStack) 
```

It's a bit hard to read right now.

Reply:
I've implemented this now.

Just a side note: I'm not a big fan of variables only being used on one location. Also the name will never do justice, because in reality it should be `noDataNeedAndEmptyStackAndActiveButNotCompletedQuestionSection` (-:

Reply:
@afjlambert, opinion?

Reply:
I see your point. It is however the same as the regexes earlier. Sometimes naming things makes it very descriptive / easy to recognize. A good short comment also helps of course, downside is the comment is sometimes not update with the code 😊.
Small thing related to your comment, it is no longer a bug 😊 
PR looks good 👍